`attr` fails for falsey attribute values

Hey everyone, great work on the awesome chai plugin!

The `.attr()` assertion fails when asserting on an attribute with a falsey value, such as an empty string.

Consider the following html:

```html
<article itemscope itemtype="http://schema.org/Article" itemprop="article"></article>
```

Asserting `expect(article).to.have.attr('itemscope')` fails when using enzyme's `mount`. It fails because [ReactTestWrapper calls `getAttribute(name) || undefined`](https://github.com/producthunt/chai-enzyme/blob/master/src/ReactTestWrapper.js#L31-L33) - and the `itemscope`'s result from `getAttribute` is `''`.

It should be noted that (although I haven't actually tested) - I believe that the CheerioWrapper won't do the same ([it doesn't use `|| undefined` in the wrapper code](https://github.com/producthunt/chai-enzyme/blob/master/src/CheerioTestWrapper.js#L33-L35) and [Cheerio itself returns the property verbatum](https://github.com/cheeriojs/cheerio/blob/master/lib/api/attributes.js#L38-L41)). 

### Solution

Do not use `|| undefined`. Instead pass the attribute verbatim. `getAttribute` will return `null` for attributes which don't exist on the element, which can be used to accurately detect the lack of an attribute over falsey values.

Same issue, especially when testing input fields...
```
AssertionError: expected the node in <Checkbox /> to have a 'disabled' attribute     
     
     HTML:
     
     <input type="checkbox" disabled="">
```

--- a/src/ReactTestWrapper.js
+++ b/src/ReactTestWrapper.js
@@ -29,7 +29,9 @@ export default class ReactTestWrapper extends TestWrapper {
   }
 
   attr (name) {
-    return this.el.getAttribute(name) || undefined
+    if (this.el.hasAttribute(name)) {
+      return this.el.getAttribute(name) || ''
+    }
   }
 
   html () {

--- a/test/attr.test.js
+++ b/test/attr.test.js
@@ -1,10 +1,21 @@
+const magicToStringObject = {
+  toString () {
+    return 'magic'
+  }
+}
+
 class Fixture extends React.Component {
   render () {
+    /* eslint-disable react/jsx-boolean-value */
     return (
       <div id='root'>
         <span id='child'>test</span>
+        <video itemScope allowFullScreen={true} autoPlay={''} hidden={false} controls={null} loop={undefined}>test2</video>
+        <audio role name={''} accessKey={false} spellCheck={null} rel={magicToStringObject}>test3</audio>
+        <tr rowSpan={0} rows={0} cols={'4'} size={'0'}></tr>
       </div>
     )
+    /* eslint-enable react/jsx-boolean-value */
   }
 }
 
@@ -28,6 +39,68 @@ describe('#attr', () => {
     it('passes negated when the actual does not match the expected', (wrapper) => {
       expect(wrapper).to.not.have.attr('disabled')
       expect(wrapper.find('span')).to.not.have.attr('disabled')
+    })
+
+    describe('HAS_NUMERIC_VALUE attrs', () => {
+      it('passes when attribute exists without a `0` vaue', (wrapper) => {
+        expect(wrapper.find('tr')).to.have.attr('rowspan')
+      })
+    })
+
+    describe('HAS_POSITIVE_NUMERIC_VALUE attrs', () => {
+      it('passes when attribute exists with a string value', (wrapper) => {
+        expect(wrapper.find('tr')).to.have.attr('cols')
+      })
+
+      it('passes negated when attribute exists but has value `0`', (wrapper) => {
+        expect(wrapper.find('tr')).to.not.have.attr('rows')
+      })
+
+      it('passes negated when attribute exists but has value `"0"`', (wrapper) => {
+        expect(wrapper.find('tr')).to.not.have.attr('size')
+      })
+    })
+
+    describe('HAS_BOOLEAN_VALUE attrs', () => {
+      it('passes when attribute exists without a value', (wrapper) => {
+        expect(wrapper.find('video')).to.have.attr('itemscope')
+      })
+
+      it('passes negated when attribute exists with a falsey (but not false/null/undefined) value', (wrapper) => {
+        expect(wrapper.find('video')).to.not.have.attr('autoplay')
+        expect(wrapper.find('video')).to.not.have.attr('autoPlay')
+      })
+
+      it('passes negated when attribute exists but has value `false`', (wrapper) => {
+        expect(wrapper.find('video')).to.not.have.attr('hidden')
+      })
+
+      it('passes negated when attribute exists but has value `null`', (wrapper) => {
+        expect(wrapper.find('video')).to.not.have.attr('controls')
+      })
+
+      it('passes negated when attribute exists but has value `undefined`', (wrapper) => {
+        expect(wrapper.find('video')).to.not.have.attr('loop')
+      })
+    })
+
+    describe('regular attrs', () => {
+      it('passes when attribute exists without a value', (wrapper) => {
+        expect(wrapper.find('audio')).to.have.attr('role')
+      })
+
+      it('passes when attribute exists with a falsey (but not false/null/undefined) value', (wrapper) => {
+        expect(wrapper.find('audio')).to.have.attr('name')
+      })
+
+      it('passes when attribute exists but has value `false`', (wrapper) => {
+        expect(wrapper.find('audio')).to.have.attr('accesskey')
+      })
+
+      it('passes negated when attribute exists but has value `null`', (wrapper) => {
+        expect(wrapper.find('audio')).to.not.have.attr('spellcheck')
+        expect(wrapper.find('audio')).to.not.have.attr('spellCheck')
+      })
     })
 
     it('fails when the actual does not match the expected', (wrapper) => {
@@ -60,6 +133,20 @@ describe('#attr', () => {
       expect(() => {
         expect(wrapper.find('span')).to.have.attr('id', 'invalid')
       }).to.throw("to have a 'id' attr")
+    })
+
+    describe('HAS_BOOLEAN_VALUE attrs', () => {
+      it('converts values to empty strings', (wrapper) => {
+        expect(wrapper.find('video')).to.have.attr('itemscope', '')
+        expect(wrapper.find('video')).to.have.attr('allowfullscreen', '')
+      })
+    })
+
+    describe('regular attrs', () => {
+      it('converts values to strings', (wrapper) => {
+        expect(wrapper.find('audio')).to.have.attr('accesskey', 'false')
+        expect(wrapper.find('audio')).to.have.attr('rel', 'magic')
+      })
     })
   })
 

